### PR TITLE
BenchmarkDotNet can benchmark local CoreFX builds using CoreRun

### PR DESCRIPTION
**Pull request by adamsitnik**
BDN can now run benchmarks using CoreRun and all you need to do is to provide the path to CoreRun using console argument.

```log
dotnet run -c Release -f netcoreapp2.1 -- --coreRun "C:\Projects\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\9.9.9\CoreRun.exe"
```

It simplifies everything and makes it very easy to benchmark local CoreFX builds!

Please take a look at the doc change to find out more.

/cc @jorive @stephentoub @danmosemsft @AndreyAkinshin

### CODE REVIEW

File: Documentation/project-docs/benchmarking.md [hunk 1]
```diff
@@ -1,239 +1,83 @@
...
 ```
 <ItemGroup>
-   <PackageReference Include="BenchmarkDotNet" Version="0.10.13" />
+   <PackageReference Include="BenchmarkDotNet" Version="0.11.1" />
 </ItemGroup>
 ```
 
 ## Defining your benchmark
 
 See [BenchmarkDotNet](http://benchmarkdotnet.org/Guides/GettingStarted.htm) documentation -- minimally you need to adorn a public method with the `[Benchmark]` attribute but there are many other ways to customize what is done such as using parameter sets or setup/cleanup methods. Of course, you'll want to bracket just the relevant code in your benchmark, ensure there are sufficient iterations that you minimise noise, as well as leaving the machine otherwise idle while you measure.
```
Comment (AndreyAkinshin) [inline Documentation/project-docs/benchmarking.md]: This link should be updated: `https://benchmarkdotnet.org/articles/guides/getting-started.html`
I added redirects for all "old-style" doc pages, so people will see this page anyway, but I still want to update links wherever possible.

File: Documentation/project-docs/benchmarking.md [hunk 2]
```diff
@@ -1,239 +1,83 @@
...
+Since `0.11.1` BenchmarkDotNet knows how to run benchmarks with CoreRun. So you just need to provide it the path to CoreRun! The simplest way to do that is via console line arguments:
 
-**Important:** BenchmarkDotNet will generate the right `.csproj` file for the self-contained app. It's going to reference the `.csproj` file of the project which defines benchmarks. It's going to work even if your project is not self-contained app targeting local CoreFX build. So you can just create a new solution with console app in Visual Studio, install BenchmarkDotNet and it's going to do the right thing for you. 
+    dotnet run -c Release -f netcoreapp2.1 -- -f *MyBenchmarkName* --coreRun "C:\Projects\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\9.9.9\CoreRun.exe"
 
 **Hint:** If you are curious to know what BDN does internally you just need to apply `[KeepBenchmarkFiles]` attribute to your class or set `KeepBenchmarkFiles = true` in your config file. After runing the benchmarks you can find the auto-generated files in `%pathToBenchmarkApp\bin\Release\$TFM\` folder.
```
Comment (danmosemsft) [inline Documentation/project-docs/benchmarking.md]: Typo....'runing'

File: Documentation/project-docs/benchmarking.md [hunk 3]
```diff
@@ -1,239 +1,83 @@
...
-# Benchmarking ANY CoreCLR and CoreFX builds
+After that, you should be able to find `CoreRun.exe` in a location similar to:
 
-BenchmarkDotNet allows you to benchmark **ANY** CoreCLR and CoreFX builds. It just generates the right `.csproj` file with appropriate dependencies and `NuGet.config` file with the right feeds.
+        C:\Projects\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\9.9.9\CoreRun.exe
 
-Example:
+2. Create a new .NET Core console app using your favorite IDE
+3. Install BenchmarkDotNet (0.11.1+)
+4. Define the benchmarks
```
Comment (danmosemsft) [inline Documentation/project-docs/benchmarking.md]: Missing #1?

File: Documentation/project-docs/benchmarking.md [hunk 4]
```diff
@@ -1,239 +1,83 @@
...
-using BenchmarkDotNet.Columns;
-using BenchmarkDotNet.Configs;
-using BenchmarkDotNet.Diagnosers;
-using BenchmarkDotNet.Environments;
-using BenchmarkDotNet.Exporters;
-using BenchmarkDotNet.Jobs;
-using BenchmarkDotNet.Loggers;
-using BenchmarkDotNet.Toolchains.CsProj;
-using BenchmarkDotNet.Toolchains.DotNetCli;
+9. Run the benchmarks using `--coreRun` from the fist step. Save the results in a dedicated folder.
```
Comment (danmosemsft) [inline Documentation/project-docs/benchmarking.md]: Typo

File: Documentation/project-docs/benchmarking.md [hunk 5]
```diff
@@ -1,239 +1,83 @@
...
-static void Main(string[] args)
-    => BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly)
-        .Run(args, DefaultConfig.Instance
-            .With(Job.ShortRun
-                .With(CustomCoreClrToolchain.CreateForNightlyCoreFxBuild("4.5.0-preview2-26215-01"))));
-```
+# Recommended workflow
 
-**Hint:** If you would like to compare the performance of different CoreFX versions, you just need to define multiple jobs, each using it's own toolchain.
+1. Before you start benchmarking the code you need to build entire CoreFX in Release which is going to generate the right CoreRun bits for you:
```
Comment (adamsitnik) [inline Documentation/project-docs/benchmarking.md]: @danmosemsft the missing `#1` is here

**Pull request merged**